### DISCUSSION TITLE
The cost of introducing this framework to one's system

### DISCUSSION THREAD

**Discussion by clean99** [autometrics-dev/autometrics-shared#71] [Ideas]
It is not clear what is the cost to introduce this framework to the system, can we add some metrics(like memory usage, time consuming, so on.) to tell other developers about the cost, so that they can better decide on use it or not to use it? And it is also a good way(data driven) to observe whether the framework becomes better or worsen.

**Comment by emschwartz** [comment]
Good suggestion! We added some CPU benchmarks for the Rust implementation [here](https://github.com/autometrics-dev/autometrics-rs#benchmarks). Memory usage is also a good one, though I'm not as sure how to set up that test.
